source/agora/common/TransactionPool.d

```diff
@@ -81,8 +81,8 @@ public class TransactionPool
     /// SQLite db instance
     private ManagedDatabase db;
 
-    /// Map for hashes for Input objects of each transaction in TX pool
-    private Set!Hash input_set;
+    /// Keeps track of which TXs spend which inputs
+    private Hash[][Hash] spenders;
```

Comment:
Shouldn't this still be a `Set!Hash[Hash]`? We don't want to add duplicate transactions (meaning ones with the exact same hash).

Reply:
Well I guess `isValidTransaction` already handles duplicate txs. You can keep it as an array if it makes the code simpler I suppose.

Reply:
The duplicate TX addition has very little to do with the `spenders` structure, the main storage is the DB. But I dont think `Set!Hash[Hash]` would make the code any more complex and would better represent the intention. I will give it a try :+1:

Reply:
It even made it a touch simpler :) Thanks @AndrejMitrovic